src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -170,8 +169,6 @@ def _requires_distributed_sampler(self, dataloader: DataLoader) -> bool:
             and self.trainer._accelerator_connector.is_distributed
             and not isinstance(dataloader.sampler, DistributedSampler)
             and not has_iterable_dataset(dataloader)
-            # `DistributedSampler` is never used with `poptorch.DataLoader`
-            and not isinstance(self.trainer.accelerator, IPUAccelerator)
```

Comment:
Why can we delete these lines? Aren't they needed? I'm pretty sure that the external implementation can't handle this.

Reply:
reverted plus the one bellow too